frame/dapps-staking/src/lib.rs

```diff
@@ -80,8 +80,22 @@ pub type BalanceOf<T> =
 /// Counter for the number of eras that have passed.
 pub type EraIndex = u32;
 
+// This represents the max assumed vector length that any storage item should have. In particular, this relates to `UnbondingInfo` and `StakerInfo`.
+// In structs which are bound in size, `MaxEncodedLen` can just be derived but that's not the case for standard `vec`.
+// To fix this 100% correctly, we'd need to do one of the following:
+//
+// - Use `BoundedVec` instead of `Vec` and do storage migration
+// - Introduce a new type `S: Get<u32>` into the aforementioned structs and use it to inject max allowed size, thus allowing us to correctly calculate max encoded len
+//
+// The issue with first approach is that it requires storage migration which we want to avoid unless it's really necessary.
+// The issue with second approach is that it makes code much more difficult to work with since all of it will be ridden with injections of the `S` type.
+//
+// Since dApps staking has been stable for long time and there are plans to redesign & refactor it, doing neither of the above makes sense, timewise.
+// So we use an assumption that Vec length won't go over the following constant.
```

Comment:
Do we have max line length specified somewhere in our code guidelines? I believe the format check should handle this.

Reply:
Not sure whether that's specified anywhere.

Format checker can indeed check this but we're using default settings for formatting.
- https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#comment_width
- https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#wrap_comments

Comment wrapping is disabled by default.